src/client/features/interactions/cy/index.js

```diff
@@ -68,13 +68,41 @@ let bindEvents = cy => {
   cy.on('layoutstart', () => hideTooltips());
   cy.on('hide-type', () => hideTooltips());
   cy.on('slider-change', () => hideTooltips());
+
+  let nodeHoverMouseOver = _.debounce(evt => {
+    let node = evt.target;
+    let elesToHighlight = cy.collection();
+
+    //Create a list of the hovered node & its neighbourhood
+    node.neighborhood().nodes().union(node).forEach(node => {
+      elesToHighlight.merge(node.ancestors());
+      elesToHighlight.merge(node.descendants());
+      elesToHighlight.merge(node);
+    });
+    elesToHighlight.merge(node.neighborhood().edges());
+
+    //Add highlighted class to node & its neighbourhood, unhighlighted to everything else
+    cy.elements().addClass('unhighlighted');
+    elesToHighlight.forEach(ele => {
```

Comment:
You don't need to use `forEach()` since calling a function on all collection internally applies the operation to all elements.

E.g. `elesToHighlight.removeClass('foo').addClass('bar')`

Reply:
Batching is also a good idea for changes to a large number of elements for performance

Reply:
For any mouseover/out events, it would be good to use `_.throttle()` or `_.debounce()` for performance.